### PR TITLE
Display Function Unit Tests

### PR DESCRIPTION
**Pull request by EnderSmith**
TL;DR:
`diceRoll2.js` changes:
- CLEANED UP `toggleSaved()`:
- - REPLACED `==` with `===`;
- - REPLACED `content.saved` to `content.saveMenu`;
- - ADDED `return` value; 
- CLEANED UP `printToInnerHTML()`:
- - REMOVED `var testOutput`;
- - ADDED simple `return '';`;
- ADDED `toString()` to output on `clearSumArray()` (easier to test);
`diceRollUnitTests.js` changes:
- ADDED `console.log()` section headers;
- ADDED `UnitTest()`s for all display functions;

(RE: overburdening the `UnitTest()` function:
I tried to implement the suggested code from [the last PR](https://github.com/EnderSmith/EnderSmith.github.io/pull/28) , but I didn't understand enough about what was going on to fix the problems it was having. In the interest of time [this all has to be finished by Sunday], I just want to focus on getting the last of the Unit Tests working, so that I can build the last major piece of code [adding custom user save items] before I have to hand this in.)

### CODE REVIEW

File: diceRoll/diceRollUnitTests.js [hunk 1]
```diff
@@ -140,6 +140,52 @@ function runTests(runTF) {
...
+        if (clearScreen() !== document.getElementById('dispOut').innerHTML) {
+          clearOut = false;
+        }
+        return (clearIn && clearOut);
+      }),
+      new UnitTest('clearSumArray()', function() {
+        g.sumArray = [5, 4, 3, 2, 1];
+        g.sumIndex = 5;
+        var arr = '0,0';
+        return (clearSumArray() === arr);
```
Comment (ultimatetag) [inline diceRoll/diceRollUnitTests.js:186]: When you clear sumArray, is it supposed to reset the sumIndex?

If you are working on State (global objects, e tc), You should check them directly after running the method. I.e. check that g.sumArray is empty and g.sumIndex is reset

File: diceRoll/diceRollUnitTests.js [hunk 2]
```diff
@@ -140,6 +140,52 @@ function runTests(runTF) {
...
+          addonPass = false;
+        }
+        return (replacePass && addonPass);
+      }),
+      new UnitTest('clearScreen()', function() {
+        var clearIn = true;
+        var clearOut = true;
+        document.getElementById('dispIn').innerHTML = 'not clear in';
+        document.getElementById('dispOut').innerHTML = 'not clear out';
+        if (clearScreen() !== document.getElementById('dispIn').innerHTML) {
```
Comment (ultimatetag) [inline diceRoll/diceRollUnitTests.js:174]: After calling clearScreen, why are you testing a return value from it? Seems like you should call clearScreen and then check that the screen is clear. This is same as comment below, you are changing State (the DOM is State), so you should be checking the state by itself. (I haven't looked at clearScreen, does it return something normally that you are checking? Maybe you are overloadijng the purpose of clearScreen?)

File: diceRoll/diceRollUnitTests.js [hunk 3]
```diff
@@ -140,6 +140,52 @@ function runTests(runTF) {
         return !(subRandomIntForDice(testGlobals.sumArray_expanded).includes('d'));
       }),
     ];
+    console.log('DISPLAY FUNCTION TESTS:');
+    passFail += [
+      new UnitTest('toggleSaved()', function() {
+        var result;
+        if (g.contentStatus === content.calculator) {
```
Comment (ultimatetag) [inline diceRoll/diceRollUnitTests.js:147]: In a test, you shjould setup the state of the app to a known a state, then test it. You shjould have a test that sets the g.contentStatus to cointent.calculator, then tests toggleSaved, and then another test that sets g.contentStatus to savedMenu, then test that. Your tests should be called:

'toggleSaved() from calculator'
'toggleSaved() from savedMenu

Your tests also shouldn't preferably not change the state of the app for other tests ... You should pass your state object into the test, or reset the state at the end. You can have code that runs before and after every test to reset the state.

Comment (ultimatetag) [reply diceRoll/diceRollUnitTests.js:147]: (should preferably)

**Review by ultimatetag — APPROVED** [review]
With suggestions.
Your test framework some major changes, but i can help refactor separately

File: diceRoll/diceRollUnitTests.js (hunk 3 above)
Comment (EnderSmith) [reply diceRoll/diceRollUnitTests.js:147]: Okay, thank you.

File: diceRoll/diceRollUnitTests.js (hunk 2 above)
Comment (EnderSmith) [reply diceRoll/diceRollUnitTests.js:174]: No, I'll fix that.

File: diceRoll/diceRollUnitTests.js (hunk 1 above)
Comment (EnderSmith) [reply diceRoll/diceRollUnitTests.js:186]: Yes, because the sumIndex tracks the index for writing to sumArray. When I reset them, I always reset them together.

clearSumArray returns the state values in an array.

Comment (EnderSmith) [reply diceRoll/diceRollUnitTests.js:186]: Completing the pull request without suggested changes
